path/spanning_tree.go

```diff
@@ -5,101 +5,166 @@
 package path
 
 import (
+	"container/heap"
+	"math"
 	"sort"
 
 	"github.com/gonum/graph"
 	"github.com/gonum/graph/concrete"
-	"github.com/gonum/graph/internal"
 )
 
-// EdgeListerGraph is an undirected graph than returns its complete set of edges.
-type EdgeListerGraph interface {
+// UndirectedWeighter is an undirected graph that returns distinct edge weights.
```

Comment:
That wording didn't work with the structure of the rest of the para, but I have something close in there now.